grpc/src/test/java/com/linecorp/armeria/common/grpc/protocol/AbstractUnaryGrpcServiceTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.grpc.protocol;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.UncheckedIOException;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.grpc.GrpcHeaderNames;
+import com.linecorp.armeria.common.grpc.GrpcSerializationFormats;
+import com.linecorp.armeria.grpc.testing.Messages.Payload;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleRequest;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleResponse;
+import com.linecorp.armeria.grpc.testing.TestServiceGrpc.TestServiceBlockingStub;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.server.ServerRule;
+
+public class AbstractUnaryGrpcServiceTest {
+
+    // This service only depends on protobuf. Users can use a custom decoder / encoder to avoid even that.
```

Comment:
maybe add an example with an empty response? I'm not sure if my code is messed up or not :) 

```
io.grpc.StatusRuntimeException: UNKNOWN: HTTP status code 200
invalid content-type: null
headers: Metadata(:status=200,grpc-encoding=identity)
DATA-----------------------------

Reply:
@anuraaga @adriancole Please let me know if this PR is ready to be merged.

Reply:
Not ready - I didn't follow my package-info comment and depended on armeria-grpc's GrpcHeaderNames, which isn't armeria-core ;) Will fix it probably by inlining and moving the class.